docs/features/architecture-analysis.md

```
@@ -8,6 +8,18 @@ repository-level** dependency graph built during indexing. These commands answer
 All three commands derive from the same `SymbolReference` edges populated during
 `codesearch index`, so re-index after code changes to keep the analysis current.
 
+### Result caching
```

markdownlint-cli2 (0.22.1)

[warning] Heading levels should only increment by one level at a time
Expected: h2; Actual: h3

(MD001, heading-increment)